docs/register-api-plugins.md

```diff
@@ -0,0 +1,169 @@
+### Register API Plugins
```

Comment:
Can we merge with this one about [developing plugins](https://github.com/reactioncommerce/docs/pull/47/files#diff-88390d67a4d665d22b449837fafcbe1b20e6b42a6850af71b3d0b662f2a68bda)

Reply:
Although there is some overlap I don't think these are the same doc. Should definitely cross link though.

Reply:
Actually I take that back. I think there needs to be a separate doc about `context` (as was the original ticket) and the other content here should be merged with that doc.